openhands/server/config/openhands_config.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+from fastapi import HTTPException
+
+from openhands.server.types import AppMode, OpenhandsConfigInterface
+from openhands.utils.import_utils import import_from
+
+
+class OpenhandsConfig(OpenhandsConfigInterface):
+    config_path = os.environ.get('OPENHANDS_CONFIG_PATH', None)
+    app_mode = AppMode.OSS
+    posthog_client_key = 'phc_3ESMmY9SgqEAGBB6sMGK5ayYHkeUuknH2vP6FmWH9RA'
+    github_client_id = os.environ.get('GITHUB_APP_CLIENT_ID', '')
+    attach_session_middleware_path = (
+        'openhands.server.middleware.AttachSessionMiddleware'
+    )
+
+    def verify_config(self):
+        if self.config_path:
+            raise ValueError('Unexpected config path provided')
+
+    def verify_github_repo_list(self, installation_id: int | None):
+        if self.app_mode == AppMode.OSS and installation_id:
+            raise HTTPException(
```

Comment:
Interestingly, now "AppMode.OSS" is only used around here. In the config class that's only instantiated for the open source project to begin with.

That suggests it might not be necessary in the server. As you accurately pointed out below, if we're running this code, then it's not saas. 😇

We can keep it here (in config) as a sanity check of sorts.

Hey it could also be called AppMode.DEFAULT or "jinglebell". 😂

Reply:
I think having this as sanity check is helpful! It has come in handy during development personally

Also the frontend still needs information on AppMode (`get_config`) which I'd like to centralize under one config server-side